cpg_workflows/targets.py

```diff
@@ -97,18 +97,118 @@ def rich_id_map(self) -> dict[str, str]:
+
+    def __repr__(self):
+        return f'MultiCohort({len(self.get_cohorts())} cohorts)'
+
+    @property
+    def target_id(self) -> str:
+        """Unique target ID"""
+        return self.name
+
+    def get_cohorts(self, only_active: bool = True) -> list['Cohort']:
+        """
+        Gets list of all cohorts.
+        Include only "active" cohorts (unless only_active is False)
+        """
+        cohorts = [cohort for _name, cohort in self._cohorts_by_name.items()]
+        if only_active:
+            cohorts = [c for c in cohorts if c.active and c.get_datasets()]
+        return cohorts
+
+    def get_cohort_by_name(self, name: str, only_active: bool = True) -> Optional['Cohort']:
+        """
+        Get cohort by name.
+        Include only "active" cohorts (unless only_active is False)
+        """
+        cohort = self._cohorts_by_name.get(name)
+        if not cohort:
+            logging.warning(f'Cohort {name} not found in the multi-cohort')
+            return None
+        if not only_active:  # Return cohort even if it's inactive
+            return cohort
+        if cohort.active and cohort.get_datasets():
+            return cohort
+        return None
+
+    def get_datasets(self, only_active: bool = True) -> list['Dataset']:
+        """
+        Gets list of all datasets.
+        Include only "active" datasets (unless only_active is False)
+        """
+        datasets = []
+        for cohort in self.get_cohorts(only_active):
+            datasets.extend(cohort.get_datasets(only_active))
+        return datasets
+
+    def get_sequencing_groups(self, only_active: bool = True) -> list['SequencingGroup']:
+        """
+        Gets a flat list of all sequencing groups from all cohorts.
+        Include only "active" sequencing groups (unless only_active is False)
+        """
+        all_sequencing_groups = []
+        for cohort in self.get_cohorts(only_active):
+            all_sequencing_groups.extend(cohort.get_sequencing_groups(only_active))
+        return all_sequencing_groups
+
+    def create_cohort(
+        self,
+        name: str,
+    ):
+        """
+        Create a cohort and add it to the multi-cohort.
+        """
+        if name in self._cohorts_by_name:
+            logging.debug(f'Cohort {name} already exists in the multi-cohort')
+            return self._cohorts_by_name[name]
+
+        c = Cohort(name=name, multicohort=self)
+        self._cohorts_by_name[c.name] = c
+        return c
+
+    def get_job_attrs(self) -> dict:
+        """
+        Attributes for Hail Batch job.
+        """
+        return {
+            'sequencing_groups': self.get_sequencing_group_ids(),
+            'datasets': [d.name for d in self.get_datasets()],
+            'cohorts': [c.name for c in self.get_cohorts()],
+        }
+
+
 class Cohort(Target):
     """
     Represents a "cohort" target - all sequencing groups from all datasets in the workflow.
```

Comment:
```suggestion
    Represents a "cohort" target - all sequencing groups from a single CustomCohort (potentially spanning multiple datasets) in the workflow.
```
Not fully confident on this, but seems like this is now out of sync. Should this be 'all SG IDs in a single custom cohort, unless only datasets are used as inputs, in which case... it represents all SG IDs, but this use case is being phased out?'

Reply:
Your correction is right! As is your understanding.